legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/ingestmode/UnitemporalSnapshotAbstract.java

```diff
@@ -55,4 +56,24 @@ default <T> T accept(IngestModeVisitor<T> visitor)
     {
         return visitor.visitUnitemporalSnapshot(this);
     }
+
+    @Value.Check
+    default void validate()
+    {
+        // All the keys in partitionValuesByField must be present in partitionFields
```

Comment:
The phrasing of the comment may be a little misleading, as it sounds like it allows partitionFields to be a super set of partitionValuesByField. maybe we can say "All the keys in partitionValuesByField must exactly match the fields in partitionFields"

Reply:
Changed as suggested